samples/saga/batching/Core_7/WorkProcessor/WorkProcessingHandler.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NServiceBus;
+
+public class WorkProcessingHandler : IHandleMessages<ProcessWorkOrder>
+{
+    public async Task Handle(ProcessWorkOrder message, IMessageHandlerContext context)
+    {
+        Console.WriteLine($"Processing work order '{message.WorkOrder}'");
+
+        try
+        {
+            PerformJob();
+
+            await context.Reply(new WorkOrderCompleted
+            {
+                ProcessId = message.ProcessId,
+                WorkOrderNo = message.WorkOrder,
+                Status = WorkStatus.Success
+            });
+        }
+        catch (Exception ex)
+        {
+            await context.Reply(new WorkOrderCompleted
```

Comment:
This would result in committing any existing transaction which is tricky. Likely its better to still throw but to do an immediate send for the reply and to throw an Exception that will cause the message to be either discarded or moved to the error queue.

Its pretty tricky.

Reply:
The idea was that the receiver can decide to retry the task, continue with the rest of the tasks, or abort. This is a design / business decision and to me, depending on what you want to do, all three are valid.